src/analysis/analytic/uniqueness.lean

```diff
@@ -103,4 +103,14 @@ begin
     λ z hz, (hf.sub hg).eq_on_zero_of_preconnected_of_eventually_eq_zero hU h₀ hfg' hz,
 end
 
+/-- The *identity principle* for analytic functions: If two analytic functions on a normed space
+coincide in a neighborhood of a point `z₀`, then they coincide everywhere.
+For a one-dimensional version assuming only that the functions coincide at some points
+arbitrarily close to `z₀`, see `eq_of_frequently_eq`. -/
+theorem eq_of_eventually_eq {f g : E → F} [preconnected_space E]
```

Comment:
Would it make more sense to assume that the field itself is connected? If you have a vector space over C, say, then it is automatically connected but the typeclass inference system can not guess it because it can not go looking for an unknown field of scalars. While connectedness of the field is something it can find by itself.

Reply:
I haven't found a good route to a proof only assuming `is_connected 𝕜`. Would you prefer having `[normed_space ℝ E]` as the hypothesis?

Reply:
(or just `[module ℝ E] [has_continuous_smul ℝ E]` to avoid insisting on a choice of norm)

Reply:
In fact we already have an instance that says that a vector space over the reals (or the complexes) is connected, so let's keep it that way, it should be good enough for applications.